### PR TITLE
Fix requirements for min and max amount

### LINKED ISSUE #142: [S3A Low] `maxAmount` should be strictly greater than `minAmount`
### Context

https://github.com/Secure3Audit/findings_NeoX_Bridge_Contract/issues/64

### PR DESCRIPTION
**Pull request by mialbu**
Fixes #142

Additionally to #142, a redundant check was removed, since the case is already covered in the underlying call to `_isValidConfig()`.

### CODE REVIEW

File: contracts/bridge/BridgeStorage.sol [hunk 1]
```diff
@@ -253,7 +253,7 @@ abstract contract BridgeStorage is BridgeStorageV1, UUPSUpgradeable {
         address _neoXToken,
         uint256 _amount
     ) internal {
-        if (_amount > tokenBridges[_neoXToken].config.maxAmount)
+        if (_amount >= tokenBridges[_neoXToken].config.maxAmount)
```
Comment (txhsl) [inline contracts/bridge/BridgeStorage.sol:256]: This means the `_amount` can not equal `maxAmount`, which seems not user-friendly, and it doesn't help #142. Is it what we expect?

Comment (mialbu) [reply contracts/bridge/BridgeStorage.sol:256]: How does it not help #142?

Check out the S3A issue [here](https://github.com/Secure3Audit/findings_NeoX_Bridge_Contract/issues/64).

If we have a `min` and a `max` I agree with the issue in the sense that it's generally not good practice to allow them to be equal. Also, the behavior in the `registerToken()` function was already restricting it (as well as the corresponding functions on the N3 bridge contract), while the setters for the min and max amounts had other logic, i.e., allowing the values to be equal. This means that the current logic in the setters is inconsistent with `registerToken()` and the logic on the N3 side.

Comment (mialbu) [reply contracts/bridge/BridgeStorage.sol:256]: I'm not completely against allowing them to be equal, but **if** we were to allow them to be equal, we need to change the logic consistently in the registration, the setters and on the N3 side.

Comment (mialbu) [reply contracts/bridge/BridgeStorage.sol:256]: Okay, just checked, the N3 side misses the check entirely in the setters. 😂 I'll open an issue to fix it.

Comment (txhsl) [reply contracts/bridge/BridgeStorage.sol:256]: > How does it not help https://github.com/bane-labs/bridge-evm-contracts/issues/142?
> Check out the S3A issue https://github.com/Secure3Audit/findings_NeoX_Bridge_Contract/issues/64.

My bad, it helps.

Comment (mialbu) [reply contracts/bridge/BridgeStorage.sol:256]: Opened an issue for it: https://github.com/bane-labs/bridge-neo-contracts/issues/109.

**Review by txhsl — APPROVED** [review]

**Pull request merged**